iceoryx_dds/include/iceoryx_dds/gateway/dds_to_iox.hpp

```diff
@@ -38,13 +37,30 @@ class DDS2IceoryxGateway : public gateway_t
 
   public:
     DDS2IceoryxGateway() noexcept;
+
+    /// @brief Set DDS to iceoryx gateway object
+    /// @param[in] channelFactory the channel factory to propagate data across
     DDS2IceoryxGateway(ChannelFactory channelFactory) noexcept;
+
+    /// @brief Load the provided gateway configuration; a channel is setup if required (no error check on channel setup)
+    /// @param[in] config the gateway configuration to load
     void loadConfiguration(const config::GatewayConfig& config) noexcept;
+
+    /// @brief Discover messages coming from DDS.
```

Comment:
Why is the  `CaproMessages` origin in DDS?
What discovery messages? I am somehow lost and do not understand the doxygen comment at all.

I think `CaproMessages` are used only internally by RouDi.

Reply:
As discussed earlier it's now documented in the base class.

@ithier Can you answer the `CaproMessages` as you're the author?!

Reply:
I'm not quite sure, but I think it is needed for service discovery and to suppress some services like introspection. At least in the `iox_to_dds.inl` it's used that way.

Reply:
@marthtz @elBoberido Sorry I literally just saw this. I don't understand the question though. Where is the `CaproMessage` ?

Reply:
@ithier it's below the line of the reviewed comment -> `void discover(const capro::CaproMessage& msg) noexcept;`

Reply:
@elBoberido @ithier @elfenpiff  Ok, So, is it fine for now or do you want me to change anything to finish this PR?